client/ConnectionThread.py

```diff
@@ -0,0 +1,109 @@
+import threading
+import socket
+import time
+
+
+class ConnectionThread(threading.Thread):
+    def __init__(self, thread_status):
+        threading.Thread.__init__(self)
+        self.handler_status = thread_status
+        self.socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self.conn = None
+        self.lock = threading.Lock()
+
+    def run(self):
+        while not self.handler_status.to_kill:
+            if not self.handler_status.is_connected:
+                if self.handler_status.is_active_connect:
+                    self.connect()
+                else:
+                    self.listen()
+            else:
+                break
+        self.conn.settimeout(3.0)
+        while not self.handler_status.to_kill:
+            self.send_counter()
+            time.sleep(1.0)
```

Comment:
what happens if program has unexpected exit here (ie. Keyboard interrupt, SystemExit raises)?

Reply:
I think that's not handled at the moment. Need to change that.